src/components/middle/immedia/Immedia.tsx

```diff
@@ -229,164 +264,124 @@ const Immedia: FC<OwnProps & StateProps> = ({ chatId, currentUser }) => {
+
+  const sendUpdate = useCallback(() => {
+    const currentMessageId = messageId + 1;
+    setMessageId(currentMessageId);
+    const message = {
+      msgId: currentMessageId,
+      id: userId,
+      type: 'app',
+      room: roomId,
+      data: {
+        type: 'update',
+        data: {
+          image: lastSnapshot,
+          timestamp: new Date().getTime(),
+          nickname,
           id: userId,
-          type: 'app',
-          room: roomId,
-          data: {
-            type: 'update',
-            data: {
-              image: lastSnapshot,
-              timestamp: new Date().getTime(),
-              nickname,
-              id: userId,
-            },
-          },
-        };
-        console.log(INIT, 'Updating with message: ', message);
-        wsRef.current.send(JSON.stringify(message));
-      }
+        },
+      },
     };
-    if (lastSnapshot !== undefined && userId !== undefined) {
-      let updateInterval: NodeJS.Timeout;
-      if (awareness) {
-        updateInterval = setInterval(sendUpdate, UPDATE_RATE);
-      }
-      return () => clearInterval(updateInterval);
-    }
-    return undefined;
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [awareness, userId, messageId, nickname, roomId]);
+    // eslint-disable-next-line no-console
+    console.log(INIT, 'Updating with message: ', message);
+    wsRef.current?.send(JSON.stringify(message));
+  }, [userId, messageId, roomId, lastSnapshot, nickname]);
+
+  useInterval(() => {
+    sendUpdate();
+  }, awareness && connected && userId ? UPDATE_RATE : undefined);
 
   // TODO: Define the Heartbeat function to keep track user connection.
   // Keep Track of connection status
-  useEffect(() => {
-    const ping = () => {
-      const type = 'ping';
-      if (wsRef.current) {
-        const currentMessageId = messageId + 1;
-        setMessageId(currentMessageId);
-        const message = {
-          msgId: currentMessageId,
-          type,
-          room: roomId,
-          data: undefined,
-        };
-        console.log(INIT, `${type.toUpperCase()}: `, message);
-        wsRef.current.send(JSON.stringify(message));
-      }
-    };
-
-    if (roomId !== undefined && userId !== undefined) {
-      let pingInterval: NodeJS.Timeout;
-      if (awareness) {
-        console.log(INIT, 'RUNNING PING INTERVAL EVERY ', PING_RATE);
-        pingInterval = setInterval(ping, PING_RATE);
-      }
-      return () => clearInterval(pingInterval);
+  const ping = () => {
+    const type = 'ping';
+    if (wsRef.current) {
+      const currentMessageId = messageId + 1;
+      setMessageId(currentMessageId);
+      const message = {
+        msgId: currentMessageId,
+        type,
+        room: roomId,
+        data: undefined,
+      };
+      // eslint-disable-next-line no-console
+      console.log(INIT, `${type.toUpperCase()}: `, message);
+      wsRef.current.send(JSON.stringify(message));
```

Comment:
If there's no connection, this will fail. What about using `wsRef.current?.send` like we do in `sendUpdate`?

Mainly for consistency, as this shouldn't be run if we are not `connected`.

Reply:
Done in 5c13275.